### ISSUE TITLE
Ubuntu22.04 - Warning: could not connect to a running Ollama instance

### ISSUE THREAD

**Issue by ACodingfreak** [ollama/ollama#6369]
### What is the issue?

```
$ ollama --version 
Warning: could not connect to a running Ollama instance
Warning: client version is 0.3.4

$  systemctl status ollama
● ollama.service - Ollama Service
     Loaded: loaded (/etc/systemd/system/ollama.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/ollama.service.d
             └─override.conf
     Active: active (running) since Wed 2024-08-14 19:56:09 PDT; 35min ago
   Main PID: 772316 (ollama)
      Tasks: 17 (limit: 76755)
     Memory: 1.1G
        CPU: 6.959s
     CGroup: /system.slice/ollama.service
             └─772316 /usr/local/bin/ollama serve

$ cat /etc/systemd/system/ollama.service
[Unit]
Description=Ollama Service
After=network-online.target

[Service]
Environment="OLLAMA_HOST=10.10.26.188"
Environment="OLLAMA_MODELS=/opt/ollama/models/"
ExecStart=/usr/local/bin/ollama serve
User=ollama
Group=ollama
Restart=always
RestartSec=3
Environment="PATH=/usr/local/cuda-12.3/bin:/usr/local/cuda-12.3/bin:/usr/local/cuda-12.3/bin:/home/codingfreak/anaconda3/envs/llama3_test/bin:/home/codingfreak/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"

[Install]
WantedBy=default.target

```

Then I upgraded the same as shown below 
```

$ curl -fsSL https://ollama.com/install.sh | sh
>>> Downloading ollama...
######################################################################## 100.0%##O#-#                                               ######################################################################## 100.0%
>>> Installing ollama to /usr/local/bin...
>>> Adding ollama user to render group...
>>> Adding ollama user to video group...
>>> Adding current user to ollama group...
>>> Creating ollama systemd service...
>>> Enabling and starting ollama service...
>>> NVIDIA GPU installed.
$ sudo systemctl daemon-reload
$ sudo systemctl enable ollama

$ ollama --version 
Warning: could not connect to a running Ollama instance
Warning: client version is 0.3.6

$ sudo systemctl status ollama
● ollama.service - Ollama Service
     Loaded: loaded (/etc/systemd/system/ollama.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/ollama.service.d
             └─override.conf
     Active: active (running) since Wed 2024-08-14 20:39:16 PDT; 6min ago
   Main PID: 785243 (ollama)
      Tasks: 16 (limit: 76755)
     Memory: 1.1G
        CPU: 6.766s
     CGroup: /system.slice/ollama.service
             └─785243 /usr/local/bin/ollama serve

Aug 14 20:39:16 gpu01 systemd[1]: Started Ollama Service.
Aug 14 20:39:16 gpu01 ollama[785243]: 2024/08/14 20:39:16 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEV>
Aug 14 20:39:16 gpu01 ollama[785243]: time=2024-08-14T20:39:16.849-07:00 level=INFO source=images.go:782 msg="total blobs: 0"
Aug 14 20:39:16 gpu01 ollama[785243]: time=2024-08-14T20:39:16.850-07:00 level=INFO source=images.go:790 msg="total unused blobs re>
Aug 14 20:39:16 gpu01 ollama[785243]: time=2024-08-14T20:39:16.850-07:00 level=INFO source=routes.go:1172 msg="Listening on 10.10.2>
Aug 14 20:39:16 gpu01 ollama[785243]: time=2024-08-14T20:39:16.850-07:00 level=INFO source=payload.go:30 msg="extracting embedded f>
Aug 14 20:39:19 gpu01 ollama[785243]: time=2024-08-14T20:39:19.216-07:00 level=INFO source=payload.go:44 msg="Dynamic LLM libraries>
Aug 14 20:39:19 gpu01 ollama[785243]: time=2024-08-14T20:39:19.216-07:00 level=INFO source=gpu.go:204 msg="looking for compatible G>
Aug 14 20:39:19 gpu01 ollama[785243]: time=2024-08-14T20:39:19.298-07:00 level=INFO source=types.go:105 msg="inference compute" id=>

```

### OS

Linux

### GPU

Nvidia

### CPU

Intel

### Ollama version

0.3.4

**Comment by ACodingfreak** [comment]
But if I run the serve command as shown below it works properly

```
$ sudo su - ollama -s /bin/bash -c '/usr/local/bin/ollama serve'
[sudo] password for codingfreak: 
2024/08/14 21:29:57 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: OLLAMA_DEBUG:false OLLAMA_FLASH_ATTENTION:false OLLAMA_HOST:http://127.0.0.1:11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:/usr/share/ollama/.ollama/models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR: OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
time=2024-08-14T21:29:57.625-07:00 level=INFO source=images.go:782 msg="total blobs: 5"
time=2024-08-14T21:29:57.643-07:00 level=INFO source=images.go:790 msg="total unused blobs removed: 0"
time=2024-08-14T21:29:57.644-07:00 level=INFO source=routes.go:1172 msg="Listening on 127.0.0.1:11434 (version 0.3.6)"
time=2024-08-14T21:29:57.644-07:00 level=WARN source=assets.go:89 msg="process still running, skipping" pid=2236 path=/tmp/ollama719788881/ollama.pid
time=2024-08-14T21:29:57.644-07:00 level=INFO source=payload.go:30 msg="extracting embedded files" dir=/tmp/ollama1752088433/runners
time=2024-08-14T21:29:59.957-07:00 level=INFO source=payload.go:44 msg="Dynamic LLM libraries [rocm_v60102 cpu cpu_avx cpu_avx2 cuda_v11]"
time=2024-08-14T21:29:59.957-07:00 level=INFO source=gpu.go:204 msg="looking for compatible GPUs"
time=2024-08-14T21:30:00.029-07:00 level=INFO source=types.go:105 msg="inference compute" id=GPU-838e39e8-08a3-37c0-85d9-66182a7927ba library=cuda compute=8.6 driver=12.2 name="NVIDIA GeForce RTX 3060" total="11.8 GiB" available="11.6 GiB"

$ ollama --version 
ollama version is 0.3.6
```

**Comment by rick-github** [comment]
> Environment="OLLAMA_HOST=10.10.26.188"

The server is configured to bind to a specific IP address, the client uses a default of `localhost`.  I'm assuming this has been set to allow connections from different devices.  Either set this to `OLLAMA_HOST=0.0.0.0` in ollama.service, or add `export OLLAMA_HOST=10.10.26.188` to your .bashrc.

**Comment by ACodingfreak** [comment]
Well this is the ipaddress of the server i.e. 10.10.26.188.

 I have modified the ipaddress to 0.0.0.0 and restarted ollama.service

```
Aug 15 06:14:21 gpu01 systemd[1]: ollama.service: Consumed 8.129s CPU time.                                                         
Aug 15 06:14:21 gpu01 systemd[1]: Started Ollama Service.                                                                           
Aug 15 06:14:21 gpu01 ollama[60589]: 2024/08/15 06:14:21 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVIC
E_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: OLLAMA_DEBUG:true OLLAMA_FLASH_ATTENTION:false OLLAMA_HOST:http://0.0.0.0:
11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODEL
S:/opt/ollama/.ollama/models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost http
s://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http:
//0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR: OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"   
```

And it works fine now.

**Comment by ouening** [comment]
I got the same error after I upgrade nvidia cuda and driver. My ollama version is 0.4.2. In ollama.service I have set 
```
Environment="OLLAMA_HOST=0.0.0.0:11434"
Environment="OLLAMA_ORIGINS=*"
```
`ollama lis`t not worked. But I can still use ollama service from remote.

**Comment by FlorianHeigl** [comment]
@ouening I had this one ollama in my container can't connect to ollama serve in the same container, listening on 0.0.0.0 and OLLAMA_HOST correctly set.., check if you had set HTTP_PROXY somewhere..
ollama is accessible from outside nonetheless. very weird. I would never have found it without your comment that it's externally accessible. I had just assumed it were broken - obviously.

## tl;dr: proxy envvars mislead ollama client
- if you use a proxy
- and have have ollama set to listen on 0.0.0.0
- you must add 0.0.0.0/32 to the NO_PROXY envvar


## proof

```
root@ollama:/# export NO_PROXY='*.my domain.tld,10.0.2.0/24,192.168.123.0/24,127.0.0.0/8'
root@ollama:/# ollama pull gemma4:26b
Error: something went wrong, please see the ollama server logs for details
root@ollama:/# export NO_PROXY='*.my.domain.tld,10.0.2.0/24,192.168.123.0/24,127.0.0.0/8,0.0.0.0/32'
root@ollama:/# ollama pull gemma4:26b
pulling manifest
pulling dfd98d273421:   0% ▕                                                                                                                                      ▏  38 MB/ 16 GB
```

and yes, that was f***ing hard to find because all those code paths don't log any error (it tells you to look at the server log, but it's a **client** error, thus nothing happens in the server logs. the server is fine... 
you can figure it out if you strace the ollama server process and nothing happens when you try to access. but it's not intuitive,  with some brain fog i never connected that, but somehow I still found it. reminder to self to bill more when I'm back at this professionally ;-)
